src/test/fakes.rs

```diff
@@ -291,18 +301,29 @@ impl<'a> FakeRelease<'a> {
                 fs::write(file, data)?;
             }
 
-            let prefix = format!(
-                "{}/{}/{}/{}",
-                prefix,
-                package.name,
-                package.version,
-                target.unwrap_or("")
-            );
             log::debug!("adding directory {} from {}", prefix, path_prefix.display());
-            crate::db::add_path_into_database(&storage, &prefix, path_prefix)
+            if archive_storage {
+                let archive = format!(
+                    "{0}/{1}/{0}-{1}-{2}.zip",
```

Comment:
Remember to update this if you change the format above. (This one likely can't be reused because you have `prefix` in the format string.)

Reply:
you're right about the duplication in code, I'll update this place too then